fix: deleted agents cannot be removed by team

carminoplata opened the pull request:
Fixes issue: [#110](https://github.com/mckinsey/agents-at-scale-ark/issues/110)
## Checklist

- [x] Follow the [contributor guide](./CONTRIBUTING.md)
- [x] End-to-end tests pass
- [x] Unit tests for essential parts of your code, e.g. APIs exposed via SDKs or endpoints
- [ ] Update `./docs`
- [ ] Recognize contributors with "@all-contributors please add <person> for <code, bug, docs, etc>"
- [x] Linked issues #eg101

Commit:
fix: showed missing agents in teams if they were deleted #110

Commit:
Merge branch 'main' into fix/deletion_of_agents_from_teams

Commit:
fix: linting errors

Commit:
Merge branch 'main' into fix/deletion_of_agents_from_teams

Commit:
test: Added unit tests for team-card component

Commit:
Merge branch 'main' into fix/deletion_of_agents_from_teams

havasik reviewed (changes requested):
Great progress @carminoplata, thanks a lot!!

https://github.com/user-attachments/assets/b89c6ef5-bebd-407e-b7a4-280eadee3b8b

Seems like removed agents are implicitly, silently removed from as team members and as team graph nodes. This can be confusing to users who'd just open up the team editor to make some unrelated changes. Let's make all edits to the team spec based on an explicit user action (clicking on the new remove agent button on team editor, clicking on remove graph edge etc.)

Commit:
fix: graph edges are not removed although team member is not present anymore
fix2: added check for missing members on selector agent too
feat: disable update button if graph or selector are invalid into team editor
feat: showed red borders on invalid selects for team editor

Commit:
fix: linting problems

Commit:
Merge branch 'main' into fix/deletion_of_agents_from_teams

dwmkerr commented:
@all-contributors please add @carminoplata for code,docs

dwmkerr commented:
thanks @carminoplata this is awesome!!

dwmkerr committed:
Merge branch 'main' into fix/deletion_of_agents_from_teams
